lstchain/io/lsteventsource.py

```diff
@@ -0,0 +1,379 @@
+        event_container.tib_masked_trigger = unpacked_tib[4]
+
+        #event_container.cdts_data = event.lstcam.cdts_data
+        event_container.swat_data = event.lstcam.swat_data
+
+        # unpack CDTS data
+        rec_fmt = '=IIIQQBBB'
+        rec_unpack = struct.Struct(rec_fmt).unpack_from
+        unpacked_cdts = rec_unpack(event.lstcam.cdts_data)
+        event_container.ucts_event_counter = unpacked_cdts[0]
+        event_container.ucts_pps_counter = unpacked_cdts[1]
+        event_container.ucts_clock_counter = unpacked_cdts[2]
+        event_container.ucts_timestamp = unpacked_cdts[3]
+        event_container.ucts_camera_timestamp = unpacked_cdts[4]
+        event_container.ucts_trigger_type = unpacked_cdts[5]
+        event_container.ucts_white_rabbit_status = unpacked_cdts[6]
+
+
+        # unpack Dragon counters
+        rec_fmt = '=HIIIQ'
+        rec_len = struct.calcsize(rec_fmt)
+        rec_unpack = struct.Struct(rec_fmt).unpack_from
+        for mod in range(self.camera_config.lstcam.num_modules):
+
+            words=event.lstcam.counters[mod*rec_len:(mod+1)*rec_len]
+            unpacked_counter = rec_unpack(words)
+            event_container.pps_counter.append(unpacked_counter[0])
+            event_container.tenMHz_counter.append(unpacked_counter[1])
+            event_container.event_counter.append(unpacked_counter[2])
+            event_container.trigger_counter.append(unpacked_counter[3])
+            event_container.local_clock_counter.append(unpacked_counter[4])
+
+        event_container.chips_flags = event.lstcam.chips_flags
+        event_container.first_capacitor_id = event.lstcam.first_capacitor_id
+        event_container.drs_tag_status = event.lstcam.drs_tag_status
+        event_container.drs_tag = event.lstcam.drs_tag
+
+    def fill_r0_camera_container_from_zfile(self, container, event):
+
+        container.num_samples = self.camera_config.num_samples
+        #container.trigger_time = event.trigger_time_s
+
+        # temporary patch to have an event time set
+        container.trigger_time = (
+            self.data.lst.tel[self.camera_config.telescope_id].evt.tib_pps_counter +
+            self.data.lst.tel[self.camera_config.telescope_id].evt.tib_tenMHz_counter * 10**(-7))
+
+        container.trigger_type = event.trigger_type
+
+        # verify the number of gains
+        if event.waveform.shape[0] == (self.camera_config.num_pixels *
+                                       container.num_samples):
+            n_gains = 1
+        elif event.waveform.shape[0] == (self.camera_config.num_pixels *
+                                         container.num_samples * 2):
+            n_gains = 2
+        else:
+            raise ValueError("Waveform matrix dimension not supported: "
+                             "N_chan x N_pix x N_samples= '{}'"
+                             .format(event.waveform.shape[0]))
+
+        container.pixel_status = np.zeros([self.n_camera_pixels])
```

Comment:
This line fails.
Currently `type(self.data) == lstchain.io.containers.LSTDataContainer` (line 280)
However `LSTDataContainer` inherits from `ctapipe.io.containers.DataContainer`.
Thus `type(self.data.r0) == ctapipe.io.containers.R0Container` which does not have `pixel_status`.

Reply:
You are write, I cleaned this part of code.

Reply:
"right"